src/proc/formats-converter.cpp

```diff
@@ -0,0 +1,396 @@
+}
+
+std::shared_ptr< stream_profile_interface >
+formats_converter::clone_profile( const std::shared_ptr< stream_profile_interface > & profile ) const
+{
+    std::shared_ptr< stream_profile_interface > cloned = nullptr;
+
+    auto vsp = std::dynamic_pointer_cast< video_stream_profile >( profile );
+    auto msp = std::dynamic_pointer_cast< motion_stream_profile >( profile );
+    if( vsp )
+    {
+        cloned = std::make_shared< video_stream_profile >( platform::stream_profile{} );
+        if( ! cloned )
+            throw librealsense::invalid_value_exception( "failed to clone profile" );
+
+        auto video_clone = std::dynamic_pointer_cast< video_stream_profile >( cloned );
+        video_clone->set_dims( vsp->get_width(), vsp->get_height() );
+        std::dynamic_pointer_cast< video_stream_profile >( cloned )->set_intrinsics( [video_clone]() {
+            return video_clone->get_intrinsics();
+        } );
+    }
+    else if( msp )
+    {
+        cloned = std::make_shared< motion_stream_profile >( platform::stream_profile{} );
+        if( ! cloned )
+            throw librealsense::invalid_value_exception( "failed to clone profile" );
+
+        auto motion_clone = std::dynamic_pointer_cast< motion_stream_profile >( cloned );
+        std::dynamic_pointer_cast< motion_stream_profile >( cloned )->set_intrinsics( [motion_clone]() {
+            return motion_clone->get_intrinsics();
+        } );
+    }
+    else
+        throw librealsense::not_implemented_exception( "Unsupported profile type to clone" );
+
+    cloned->set_framerate( profile->get_framerate() );
+
+    // No need to set the ID, when calling get_all_possible_target_profiles the from_profiles don't have an assigned ID
+    // yet, it will be assigned according to stream later on.
+    // cloned->set_unique_id( profile->get_unique_id() ); // clone() issues a new id for the profile, we want to use old one
+    // 
+    // Needed for clone_profile wholeness but will be overwritten later by target profile fields. Can skip.
+    // cloned->set_format( profile->get_format() );
+    // cloned->set_stream_index( profile->get_stream_index() );
+    // cloned->set_stream_type( profile->get_stream_type() );
+
+    return cloned;
+}
+
+bool formats_converter::is_profile_in_list( const std::shared_ptr< stream_profile_interface > & profile,
+                                            const stream_profiles & profiles ) const
+{
+    // Converting to stream_profile to avoid dynamic casting to video/motion_stream_profile
+    const auto && is_duplicate_predicate = [&profile]( const std::shared_ptr< stream_profile_interface > & spi ) {
+        return to_profile( spi.get() ) == to_profile( profile.get() );
+    };
+
+    return std::any_of( begin( profiles ), end( profiles ), is_duplicate_predicate );
+}
+
+// Not passing const & because we modify target_profiles, would otherwise need to create a copy
+void formats_converter::prepare_to_convert( stream_profiles target_profiles )
```

Comment:
[Nir-Az] Was `resolve_requests`. Name changed and also separated the return value, because it now needs to return both profiles and processing blocks at several different occasions.